src/api/TourneyRent/TourneyRent.Presentation.Api/Controllers/TeamController.cs

```diff
@@ -0,0 +1,119 @@
+﻿using AutoMapper;
+using Microsoft.AspNetCore.JsonPatch;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using TourneyRent.BusinessLogic.Services;
+using TourneyRent.DataLayer;
+using TourneyRent.DataLayer.Models;
+using TourneyRent.Presentation.Api.Views.Team;
+
+namespace TourneyRent.Presentation.Api.Controllers
+{
+
+    [Route("Team")]
+    [ApiController]
+    public class TeamController : ControllerBase
+    {
+        private readonly TeamService _teamService;
+
+        private readonly IMapper _mapper;
+
+        public TeamController(TeamService teamservice, IMapper mapper)
+        {
+            _teamService = teamservice;
+            _mapper = mapper;
+        }
+
+        [HttpGet("{id}")]
+        public async Task<ActionResult<TeamRead>> GetTeamById(int id)
+        {
+            var team = await _teamService.GetTeamById(id);
+            if(team==null)
+            {
+                return NotFound();
+            }
+
+            var teamRead = _mapper.Map<TeamRead>(team);
+            return Ok(teamRead);
+        }
+
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<TeamRead>>> GetAllTeams()
+        {
+            var teams = await _teamService.GetAllTeams();
```

Comment:
Async methods should have a **Async** suffix

Reply:
As I'm seeing, most of these methods have  waiting for some response or am I understanding it wrong?

Reply:
I just meant that methods should be named like this: _teamService.GetAllTeamsAsync(), just like you used _context.SaveChangesAsync(). However, controller methods don't have to have that naming convention, since there is not going to be any consumer in the codebase.